### PR TITLE
Bring parity between eval and sim for azure_ai_project

### PR DESCRIPTION
**Pull request by nagkumar91**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py [hunk 1]
```diff
@@ -52,34 +52,31 @@ def wrapper(*args, **kwargs):
...
+        if not all(key in azure_ai_project for key in ["subscription_id", "resource_group_name", "project_name"]):
+            raise ValueError("azure_ai_project must contain keys: subscription_id, resource_group_name, project_name")
         # check the value of the keys in azure_ai_project is not none
-        if not all(
-            azure_ai_project[key] for key in ["subscription_id", "resource_group_name", "project_name", "credential"]
-        ):
-            raise ValueError("subscription_id, resource_group_name, project_name, and credential must not be None")
+        if not all(azure_ai_project[key] for key in ["subscription_id", "resource_group_name", "project_name"]):
+            raise ValueError("subscription_id, resource_group_name and project_name must not be None")
+        if "credential" in azure_ai_project and not credential:
```
Comment (singankit) [inline src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py:73]: Is this for backward compatibility ?

File: src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py [hunk 2]
```diff
@@ -52,34 +52,31 @@ def wrapper(*args, **kwargs):
...
+            credential = azure_ai_project["credential"]
         self.azure_ai_project = azure_ai_project
         self.token_manager = ManagedIdentityAPITokenManager(
             token_scope=TokenScope.DEFAULT_AZURE_MANAGEMENT,
             logger=logging.getLogger("AdversarialSimulator"),
-            credential=self.azure_ai_project["credential"],
+            credential=credential,
```
Comment (singankit) [inline src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py:79]: If credential is None what is the behavior ? Seem like earlier it was required to be part of azure_ai_project now it has default value of None

File: src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py (hunk 1 above)
Comment (nagkumar91) [reply src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py:73]: Yup.

File: src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py (hunk 2 above)
Comment (nagkumar91) [reply src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py:82]: Fixed it!

File: src/promptflow-evals/tests/evals/e2etests/test_adv_simulator.py [hunk 3]
```diff
@@ -230,3 +230,47 @@ async def callback(
...
+                max_simulation_results=1,
+                target=callback,
+                api_call_retry_limit=3,
+                api_call_retry_sleep_sec=1,
+                api_call_delay_sec=30,
+                concurrent_async_task=1,
+                jailbreak=True,
+            )
+        )
+        print(outputs.to_json_lines())
```
[the remark this answers is not in the document]
Comment (nagkumar91) [reply src/promptflow-evals/tests/evals/e2etests/test_adv_simulator.py]: Removed

**Review by singankit — APPROVED** [review]

**Review by luigiw — APPROVED** [review]

**Review by nick863 — APPROVED** [review]

**Review by singankit — APPROVED** [review]

**Pull request merged**